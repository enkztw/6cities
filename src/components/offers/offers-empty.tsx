import * as React from 'react';
import {cityName} from '../../types';


const OffersEmtpy: React.FC<{city: cityName}> = ({city}) => {
  return (
    <div className="cities">
      <div className="cities__places-container cities__places-container--empty container">
        <section className="cities__no-places">
          <div className="cities__status-wrapper tabs__content">
            <b className="cities__status">No places to stay available</b>
            <p className="cities__status-description">We could not find any property availbale at the moment in {city}</p>
          </div>
        </section>
        <div className="cities__right-section"></div>
      </div>
    </div>
  );
};

export default React.memo(OffersEmtpy);
