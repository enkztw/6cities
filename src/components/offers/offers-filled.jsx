import React from "react";
import PropTypes from 'prop-types';

import Offer from '../offer/offer.jsx';
import Map from '../map/map.jsx';

class OffersFilled extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      currentOffer: null
    };

    this.handleMouseEnter = this.handleMouseEnter.bind(this);
  }

  handleMouseEnter(e) {
    e.persist();
    e.preventDefault();


    this.setState(() => {
      return {
        currentOffer: e.target
      };
    });
  }

  render() {
    return (
      <div className="cities">
        <div className="cities__places-container container">
          <section className="cities__places places">
            <h2 className="visually-hidden">Places</h2>
            <b className="places__found">{this.props.offers.length} places to stay in {this.props.city}</b>
            <form className="places__sorting" action="#" method="get">
              <span className="places__sorting-caption">Sort by</span>
              <span className="places__sorting-type" tabIndex="0">
                Popular
                <svg className="places__sorting-arrow" width="7" height="4">
                  <use xlinkHref="#icon-arrow-select"></use>
                </svg>
              </span>
              <ul className="places__options places__options--custom">
                <li className="places__option places__option--active" tabIndex="0">Popular</li>
                <li className="places__option" tabIndex="0">Price: low to high</li>
                <li className="places__option" tabIndex="0">Price: high to low</li>
                <li className="places__option" tabIndex="0">Top rated first</li>
              </ul>
            </form>
            <div className="cities__places-list places__list tabs__content">
              {this.props.offers.map((offer) => <Offer {...offer} key={offer.id} />)}
            </div>
          </section>
          <div className="cities__right-section">
            <Map mini={true} offers={this.props.offers}/>
          </div>
        </div>
      </div>
    );
  }
}

OffersFilled.propTypes = {
  city: PropTypes.string,
  offers: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.string
  }))
};

export default OffersFilled;
