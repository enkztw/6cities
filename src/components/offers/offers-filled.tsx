import * as React from 'react';

import {Offer} from '../../types';
import withFocusedItem from '../../hocs/with-focused-item/with-focused-item';
import Card from '../offer/offer';
import Map from '../map/map';
import Sorters from '../sorters/sorters';


interface Props {
  offers: Offer[];
  city: string;
  focusedItem: number;
  onBookmarkClick: () => void;
  onMouseEnter: () => void;
  onMouseLeave: () => void;
}

const OffersFilled: React.FC<Props> = ({offers, city, focusedItem, onBookmarkClick, onMouseEnter, onMouseLeave}) => {
  return (
    <div className="cities">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <h2 className="visually-hidden">Places</h2>
          <b className="places__found">
            {offers.length} places to stay in {city}
          </b>
          <Sorters />
          <div className="cities__places-list places__list tabs__content">
            {offers.map((offer) => (
              <Card
                {...offer}
                key={offer.id}
                onBookmarkClick={onBookmarkClick}
                handleMouseEnter={onMouseEnter}
                handleMouseLeave={onMouseLeave}
              />
            ))}
          </div>
        </section>
        <div className="cities__right-section">
          <Map
            mini
            offers={offers}
            focusedOffer={focusedItem}
          />
        </div>
      </div>
    </div>
  );
};

export default withFocusedItem(OffersFilled);
