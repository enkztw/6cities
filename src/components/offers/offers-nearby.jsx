import React from 'react';
import PropTypes from 'prop-types';

import Offer from '../offer/offer.jsx';

function NearbyOffers({offers}) {
  return (
    <div className="container">
      <section className="near-places places">
        <h2 className="near-places__title">Other places in the neighbourhood</h2>
        <div className="near-places__list places__list">
          {offers.map((offer) => <Offer key={offer.id} {...offer} />)}
        </div>
      </section>
    </div>
  );
}

NearbyOffers.propTypes = {
  offers: PropTypes.array
};

export default NearbyOffers;
