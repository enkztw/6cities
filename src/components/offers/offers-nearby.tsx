import * as React from 'react';

import {Offer} from '../../types';
import Card from '../offer/offer';


interface Props {
  offers: Offer[];
}

const NearbyOffers: React.FC<Props> = ({offers}: Props) => {
  return (
    <div className="container">
      <section className="near-places places">
        <h2 className="near-places__title">Other places in the neighbourhood</h2>
        <div className="near-places__list places__list">
          {offers.map((offer) => <Card key={offer.id} {...offer} />)}
        </div>
      </section>
    </div>
  );
};

export default React.memo(NearbyOffers);
