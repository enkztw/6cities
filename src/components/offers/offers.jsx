import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {getOffers} from '../../store/reducers/offers/selectors.js';
import {getCities} from '../../store/reducers/cities/selectors.js';
import {getCity} from '../../store/reducers/cities/selectors.js';

import Tab from '../tab/tab.jsx';
import OffersFilled from './offers-filled.jsx';
import OffersEmpty from './offers-empty.jsx';

function Offers({offers, cities, city}) {
  return (
    <main className={`page__main page__main--index page__main--index-empty${offers.length > 0 ? `` : ` page__main--index-empty`}`}>
      <h1 className="visually-hidden">Cities</h1>
      <div className="tabs">
        <section className="locations container">
          <ul className="locations__list tabs__list">
            {cities.map((it) => <Tab city={it} key={it} />)}
          </ul>
        </section>
      </div>
      {offers.length > 0 ? <OffersFilled offers={offers} city={city} /> : <OffersEmpty city={city} />}
    </main>
  );
}

Offers.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    bedrooms: PropTypes.number,
    city: PropTypes.shape({
      location: PropTypes.shape({
        latitude: PropTypes.number,
        longitude: PropTypes.number,
        zoom: PropTypes.number
      }),
      name: PropTypes.string
    }),
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatarUrl: PropTypes.string,
      id: PropTypes.number,
      isPro: PropTypes.bool,
      name: PropTypes.string
    }),
    id: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    isFavorite: PropTypes.bool,
    isPremium: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number
    }),
    maxAdults: PropTypes.number,
    previewImage: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string
  })),
  cities: PropTypes.arrayOf(PropTypes.string),
  city: PropTypes.string
};

const mapStateToProps = (state) => ({
  city: getCity(state),
  cities: getCities(state),
  offers: getOffers(state)
});

export default connect(mapStateToProps)(Offers);


