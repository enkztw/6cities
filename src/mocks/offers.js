export const offers = [
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    images: [`/img/room.jpg`, `/img/apartment-01.jpg`, `/img/apartment-02.jpg`, `/img/apartment-03.jpg`, `/img/studio-01.jpg`, `/img/apartment-01.jpg`],
    bedrooms: 3,
    adults: 4,
    price: 110,
    premium: false,
    rating: 4.51,
    bookmarked: true
  },
  {
    id: 2,
    name: `Wood and stone place`,
    type: `Room`,
    images: [`/img/apartment-01.jpg`, `/img/room.jpg`, `/img/apartment-02.jpg`, `/img/apartment-03.jpg`, `/img/studio-01.jpg`, `/img/apartment-01.jpg`],
    bedrooms: 1,
    adults: 2,
    price: 120,
    premium: true,
    rating: 4.49,
    bookmarked: true
  },
  {
    id: 3,
    name: `Canal View Prinsengracht`,
    type: `House`,
    images: [`/img/apartment-02.jpg`, `/img/room.jpg`, `/img/apartment-01.jpg`, `/img/apartment-03.jpg`, `/img/studio-01.jpg`, `/img/apartment-01.jpg`],
    bedrooms: 3,
    adults: 3,
    price: 240,
    premium: false,
    rating: 2.11,
    bookmarked: false
  },
  {
    id: 4,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Hotel`,
    images: [`/img/apartment-03.jpg`, `/img/room.jpg`, `/img/apartment-01.jpg`, `/img/apartment-02.jpg`, `/img/studio-01.jpg`, `/img/apartment-01.jpg`],
    bedrooms: 1,
    adults: 2,
    price: 50,
    premium: true,
    rating: 3.2,
    bookmarked: false
  }];
