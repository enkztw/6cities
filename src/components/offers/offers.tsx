import * as React from 'react';
import {connect} from 'react-redux';

import {city} from '../../common/types';
import {Offer} from '../../common/interfaces';
import {getOffers} from '../../store/reducer/offers/selectors';
import {getCities} from '../../store/reducer/cities/selectors';
import {getCity} from '../../store/reducer/cities/selectors';

import Tab from '../tab/tab';
import OffersFilled from './offers-filled';
import OffersEmpty from './offers-empty';


interface Props {
  offers: Offer[];
  cities: string[];
  currentCity: city;
}

const Offers: React.FC<Props> = ({offers, cities, currentCity}) => {
  return (
    <main className={`page__main page__main--index page__main--index-empty${offers.length ? `` : ` page__main--index-empty`}`}>
      <h1 className="visually-hidden">Cities</h1>
      <div className="tabs">
        <section className="locations container">
          <ul className="locations__list tabs__list">
            {cities.map((name) => <Tab name={name} key={name} />)}
          </ul>
        </section>
      </div>
      {offers.length > 0 ? <OffersFilled offers={offers} city={currentCity}/> : <OffersEmpty name={currentCity} />}
    </main>
  );
};

const mapStateToProps = (state) => ({
  currentCity: getCity(state),
  cities: getCities(state),
  offers: getOffers(state)
});

export default connect(mapStateToProps)(Offers);


