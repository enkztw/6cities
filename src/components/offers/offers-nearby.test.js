import React from 'react';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {BrowserRouter as Router} from 'react-router-dom';

import OffersNearby from './offers-nearby.jsx';

Enzyme.configure({
  adapter: new Adapter()
});

import {offers} from '../../mocks/offers.js';

it(`<OffersNearby /> should render offers`, () => {
  const div = global.document.createElement(`div`);
  global.document.body.appendChild(div);
  const tree = mount(
      <Router>
        <OffersNearby offers={offers} />
      </Router>, {attachTo: div});

  expect(tree.getDOMNode()).toMatchSnapshot();
});
