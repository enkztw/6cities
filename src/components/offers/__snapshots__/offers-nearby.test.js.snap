// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OffersNearby /> should render offers 1`] = `
<div
  class="container"
>
  <section
    class="near-places places"
  >
    <h2
      class="near-places__title"
    >
      Other places in the neighbourhood
    </h2>
    <div
      class="near-places__list places__list"
    >
      <article
        class="cities__place-card place-card"
      >
        <div
          class="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="/img/room.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class="place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                110
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button button place-card__bookmark-button--active"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 100%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/1"
            >
              Beautiful & luxurious apartment at great location
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            Apartment
          </p>
        </div>
      </article>
      <article
        class="cities__place-card place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="/img/apartment-01.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class="place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                120
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button button place-card__bookmark-button--active"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 80%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/2"
            >
              Wood and stone place
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            Room
          </p>
        </div>
      </article>
      <article
        class="cities__place-card place-card"
      >
        <div
          class="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="/img/apartment-02.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class="place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                240
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button button"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 40%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/3"
            >
              Canal View Prinsengracht
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            House
          </p>
        </div>
      </article>
      <article
        class="cities__place-card place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="cities__image-wrapper place-card__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="200"
              src="/img/apartment-03.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          class="place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                50
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button button"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 60%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/4"
            >
              Nice, cozy, warm big bed apartment
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            Hotel
          </p>
        </div>
      </article>
    </div>
  </section>
</div>
`;
